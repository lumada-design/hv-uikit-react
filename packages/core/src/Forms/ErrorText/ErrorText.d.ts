import { StandardProps } from "@material-ui/core";

export interface HvErrorTextProps
  extends StandardProps<React.HTMLAttributes<HTMLDivElement>, HvInfoTextClassKey> {
  /**
   * Describes the current state of the error text.
   */
  errorTextStatus?: string;
  /**
   * The text to be shown by the error text.
   */
  label?: string;
  /**
   * When this text is to be rendered.
   */
  showWhen: "standBy" | "valid" | "invalid";
  /**
   * If ´true´ the input is disabled.
   */
  disabled?: boolean;
}

export type HvInfoTextClassKey = "errorText" | "showText" | "errorTextDisabled" | "@global";

export default function HvErrorText(props: HvErrorTextProps): JSX.Element | null;
