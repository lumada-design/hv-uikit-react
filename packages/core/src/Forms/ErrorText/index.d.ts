export { default } from "./ErrorText";
export * from "./ErrorText";
