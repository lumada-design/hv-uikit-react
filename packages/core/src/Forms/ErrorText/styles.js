const styles = theme => ({
  root: {},
  topGutter: { marginTop: 8 },
  errorTextDisabled: {
    color: theme.hv.palette.atmosphere.atmo7,
    display: "none"
  },
  errorText: {
    color: theme.hv.palette.accent.acce1,
    display: "none"
  },
  showText: {
    display: "block"
  },
  "@global": {
    "input:-webkit-autofill": {
      "-webkit-box-shadow": `0 0 0px 1000px ${theme.hv.palette.atmosphere.atmo1} inset`,
      "-webkit-text-fill-color": theme.hv.typography.normalText.color
    }
  }
});

export default styles;
