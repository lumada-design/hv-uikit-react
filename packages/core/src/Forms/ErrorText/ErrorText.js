import React from "react";
import PropTypes from "prop-types";
import clsx from "clsx";
import { withStyles } from "@material-ui/core";
import { HvFormElementContextConsumer } from "../FormElement";
import HvTypography from "../../Typography";

import styles from "./styles";
/**
 * Component used in conjunction with other form elements, to give extra information about error status.
 */
const HvErrorText = props => {
  const {
    label,
    classes,
    id,
    errorTextStatus,
    showWhen,
    disabled,
    disableGutter = false,
    ...others
  } = props;

  return (
    <HvFormElementContextConsumer>
      {formContext => {
        const { elementStatus, elementDisabled } = formContext;
        const localStatus = errorTextStatus || elementStatus || "standBy";
        const isVisible = localStatus === showWhen;
        const localDisabled = disabled || elementDisabled;
        return (
          <HvTypography
            id={id}
            variant="infoText"
            className={clsx({
              [classes.showText]: isVisible,
              [classes.errorTextDisabled]: localDisabled,
              [classes.errorText]: !localDisabled,
              [classes.topGutter]: !disableGutter
            })}
            aria-live="polite"
            aria-atomic="true"
            aria-relevant="additions text"
            {...others}
          >
            {label}
          </HvTypography>
        );
      }}
    </HvFormElementContextConsumer>
  );
};

HvErrorText.propTypes = {
  /**
   * Describes the current state of the error text.
   */
  errorTextStatus: PropTypes.string,
  /**
   * Id to be applied to the root node.
   */
  id: PropTypes.string,
  /**
   * A Jss Object used to override or extend the component styles applied.
   */
  classes: PropTypes.shape({
    /**
     * Styles applied to the information text.
     */
    errorText: PropTypes.string,
    /**
     * Styles applied when the text should be shown.
     */
    showText: PropTypes.string,
    /**
     * Styles applied when the text is disabled.
     */
    errorTextDisabled: PropTypes.string,
    /**
     * Separation between text and upper element.
     */
    topGutter: PropTypes.string,
    /**
     * IE11 specific styling.
     */
    "@global": PropTypes.string
  }).isRequired,
  /**
   * The text to be shown by the info text.
   */
  label: PropTypes.string,
  /**
   * When this text is to be rendered.
   */
  showWhen: PropTypes.oneOf(["standBy", "valid", "invalid"]),
  /**
   * If ´true´ the text is disabled.
   */
  disabled: PropTypes.bool,
  /**
   * If ´true´ the text won't include a top gutter.
   */
  disableGutter: PropTypes.bool
};

export default withStyles(styles, { name: "HvErrorText" })(HvErrorText);
