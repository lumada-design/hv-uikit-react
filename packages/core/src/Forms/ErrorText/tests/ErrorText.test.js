/* eslint-env jest */

import React from "react";
import { mount } from "enzyme";

import HvProvider from "../../../Provider";
import ErrorText from "../ErrorText";

/* eslint-disable no-console */

describe("ErrorText", () => {
  let wrapper;

  beforeEach(async () => {
    wrapper = mount(
      <HvProvider>
        <ErrorText id="base" errorTextStatus="valid" label="text" />
      </HvProvider>
    );
  });

  it("should be defined", () => {
    expect(wrapper).toBeDefined();
  });

  it("should render correctly", () => {
    expect(wrapper.find(ErrorText)).toMatchSnapshot();
  });

  it("should render the error component", () => {
    const errorTextComponent = wrapper.find(ErrorText);
    expect(errorTextComponent.length).toBe(1);
  });
});
