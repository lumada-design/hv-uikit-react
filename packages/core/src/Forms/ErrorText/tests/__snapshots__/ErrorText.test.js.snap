// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorText should render correctly 1`] = `
<WithStyles(HvErrorText)
  errorTextStatus="valid"
  id="base"
  label="text"
>
  <HvErrorText
    classes={
      Object {
        "errorText": "HvErrorText-errorText",
        "errorTextDisabled": "HvErrorText-errorTextDisabled",
        "root": "HvErrorText-root",
        "showText": "HvErrorText-showText",
        "topGutter": "HvErrorText-topGutter",
      }
    }
    errorTextStatus="valid"
    id="base"
    label="text"
  >
    <WithStyles(ForwardRef)
      aria-atomic="true"
      aria-live="polite"
      aria-relevant="additions text"
      className="HvErrorText-errorText HvErrorText-topGutter"
      id="base"
      variant="infoText"
    >
      <ForwardRef
        aria-atomic="true"
        aria-live="polite"
        aria-relevant="additions text"
        className="HvErrorText-errorText HvErrorText-topGutter"
        classes={
          Object {
            "3xlTitle": "HvTypography-3xlTitle",
            "4xlTitle": "HvTypography-4xlTitle",
            "5xlTitle": "HvTypography-5xlTitle",
            "disabledButtonText": "HvTypography-disabledButtonText",
            "highlightText": "HvTypography-highlightText",
            "infoText": "HvTypography-infoText",
            "inlineLink": "HvTypography-inlineLink",
            "lTitle": "HvTypography-lTitle",
            "labelText": "HvTypography-labelText",
            "mTitle": "HvTypography-mTitle",
            "normalText": "HvTypography-normalText",
            "placeholderText": "HvTypography-placeholderText",
            "root": "HvTypography-root",
            "sLink": "HvTypography-sLink",
            "sText": "HvTypography-sText",
            "sTitle": "HvTypography-sTitle",
            "selectedNavText": "HvTypography-selectedNavText",
            "selectedText": "HvTypography-selectedText",
            "vizText": "HvTypography-vizText",
            "xlTitle": "HvTypography-xlTitle",
            "xsTitle": "HvTypography-xsTitle",
            "xxlTitle": "HvTypography-xxlTitle",
            "xxsTitle": "HvTypography-xxsTitle",
          }
        }
        id="base"
        variant="infoText"
      >
        <p
          aria-atomic="true"
          aria-live="polite"
          aria-relevant="additions text"
          className="HvErrorText-errorText HvErrorText-topGutter HvTypography-root HvTypography-infoText"
          id="base"
        >
          text
        </p>
      </ForwardRef>
    </WithStyles(ForwardRef)>
  </HvErrorText>
</WithStyles(HvErrorText)>
`;
